File: scripts/_postgres.php
<?php

// -------------------------------------------------------------
// PostgreSQL Services Configuration
// -------------------------------------------------------------

if (empty($config['postgres'])) {
    return;
}


foreach ($config['postgres'] as $postgres) {
    $version = $postgres['version'];
    $port = $postgres['port'];
    $host = sprintf('pgsql_%s', str_replace('.', '', $version));

    $compose['services'][$host] = [
        'build' => [
            'context' => 'docker/postgres',
            'args' => ['VERSION' => $version],
        ],
        'volumes' => [
            sprintf('./var/pgsql/%s:/var/lib/postgresql/data', $version),
        ],
        'ports' => [
            sprintf('%d:5432', $port),
        ],
    ];

    $compose['services']['pgadmin']['links'][] = $host;
}

File: generator/postgres.py
def add_configuration(compose, config):
    pg_hosts = []
    for version in config['versions']:
        port = config['versions'][version]
        host = 'pgsql_'+version.replace('.', '')
        compose['services'][host] = {
            'build': {
                'context': 'docker/postgres',
                'args': { 'VERSION': version }
            },
            'volumes': [
                './var/postgres/%s:/var/lib/postgresql/data' % (version)
            ],
            'ports': [
                '%d:5432' % (port)
            ],
        }

    compose['services']['pgadmin']['links'] = host
